presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.hive.metastore.cache;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.Closer;
+import com.google.common.io.Files;
+import com.google.common.io.MoreFiles;
+import io.prestosql.Session;
+import io.prestosql.plugin.hive.HiveHadoop2Plugin;
+import io.prestosql.spi.security.Identity;
+import io.prestosql.spi.security.SelectedRole;
+import io.prestosql.testing.DistributedQueryRunner;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static io.prestosql.spi.security.SelectedRole.Type.ROLE;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@SuppressWarnings("UnstableApiUsage")   // for Closer
+@Test(singleThreaded = true)
+public class TestCachingHiveMetastoreWithQueryRunner
+{
+    private static final String CATALOG = "test";
+    private static final String SCHEMA = "test";
+    private static final Session ADMIN = getTestSession(Identity.forUser("admin")
+            .withRole(CATALOG, new SelectedRole(ROLE, Optional.of("admin")))
+            .build());
+    private static final String ALICE_NAME = "alice";
+    private static final Session ALICE = getTestSession(new Identity.Builder(ALICE_NAME).build());
+
+    private DistributedQueryRunner queryRunner;
+    private final Closer closer = Closer.create();
+
+    @BeforeMethod
+    public void createQueryRunner()
+            throws Exception
+    {
+        queryRunner = DistributedQueryRunner
```

Comment:
Use `io.prestosql.plugin.hive.HiveQueryRunner#builder` then you don't have to play with metastore directory.

Reply:
my setup is completely different from the offering of this builder, code reuse in this case makes no sense